numpy/core/src/multiarray/mapping.c

```diff
@@ -1735,36 +1725,30 @@ array_assign_subscript(PyArrayObject *self, PyObject *ind, PyObject *op)
         return -1;
     }
 
-    /* Single field access */
-    if (PyDataType_HASFIELDS(PyArray_DESCR(self))) {
-        if (PyString_Check(ind) || PyUnicode_Check(ind)) {
-            PyObject *obj;
+    /* field access */
+    if (PyDataType_HASFIELDS(PyArray_DESCR(self)) &&
+            obj_is_string_or_stringlist(ind)) {
```

Comment:
Ah, I had intended for the following two examples to behave similarly.

    >>> a = np.array([(1,2,3),(4,5,6)], dtype=[('f0', 'i8'), ('f1', 'i8'), ('f2', 'i8')])
    >>> b = a[['f0', 'f2']]
    >>> b[:] = (10,20)

    >>> a = np.array([(1,2,3),(4,5,6)], dtype=[('f0', 'i8'), ('f1', 'i8'), ('f2', 'i8')])
    >>> a[['f0', 'f2']] = (10,20)

The first example was already possible - multi-field indexing returns a copy, and shows a warning if you try to write to it. With this PR, I wanted second example to also show a warning. (Both examples effectively write to dev/null).

I forgot to show the warning though. I will add it in `array_assign_subscript`.

I don't think multi-field indexing should return a view (even as an option) yet. I tried doing so, but indeed, as you mention, I discovered that it would behave quite strangely. I think more extensive changes are needed before multi-field assignment is really allowed.